src-docs/spec/joinRealtimeSession.md

```diff
@@ -0,0 +1,69 @@
+# joinRealtimeChannel (experimental)
+
+```js
+const realtimeChannel = window.webxdc.joinRealtimeChannel();
+```
+
+Setup and return the realtime channel for this app,
+with methods for listening and sending data as well as leaving the channel. 
+Per-app realtime channels are: 
+
+- **private**: No one outside the chat can participate in realtime channels. 
+
+- **isolated**: apps can not participate in realtime channels of other apps. 
+
+- **ephemeral**: any sent data will only be received by currently
+  connected peers but not by peers connecting later.
+
+Calling `joinRealtimeChannel` a second time without leaving the prior one
+will not cause any action and returns a `null` value. 
+
+## `realtimeChannel.setListener((data) => {})` 
+
+Start listening on the realtime channel using the specified callback. 
+The callback receives `Uint8Array` data items that were sent from connected peers. 
+Calling `setListener` a second time will replace the previous listener. 
```

Comment:
what happens if the passed value is something other than a function? for instance:

```js
realtimeChannel.setListener()
```

Will that throw an error, or cause it to stop listening?

Reply:
Will do so